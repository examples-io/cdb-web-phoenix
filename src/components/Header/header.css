@media screen and (min-width: 0px) {

}

@media screen and (min-width: 640px) {

}

@media screen and (min-width: 1200px) {

}
