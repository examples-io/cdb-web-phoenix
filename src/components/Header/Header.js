import React, { Component } from 'react';
import PropTypes from 'prop-types';
import i18n from '../../utlities/language/i18nextConfig';


import './header.css';
import { AppBar, Dropdown, Image } from '../';
import scotiaLogo from '../assets/images/scotiabankLogo.svg';

/**
 * @description Header component - renders a header with optional left and right corner spaces for logos, etc.
 */
class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            language: 'en',
            languageOptions: [
                { id: 'en', text: 'EN' },
                { id: 'fr', text: 'FR' }
            ]
        }
    }

    changeLanguage(event) {
        i18n.changeLanguage(event.target.value);
    }

    render() {
        const headerLeft = (<Image alt="Scotiabank" id="scotiabankLogo" src={scotiaLogo}/>);
        const headerRight = (<Dropdown id="languagePicker" name="language" onChange={this.changeLanguage} options={this.state.languageOptions} value={this.state.language}/>);

        return (
            <AppBar
                headerLeft={headerLeft}
                headerRight={headerRight}
            />
        );
    }
}

Header.propTypes = {
    /**
     * headerLeft - optional content to be displayed in the left corner of the header
     */
    headerLeft: PropTypes.node,
    /**
     * headerRight - optional content to be displayed in the right corner of the header
     */
    headerRight: PropTypes.node,
    /**
     * children - optional content to be displayed in the center of the header
     */
    children: PropTypes.node
};

export default Header;
