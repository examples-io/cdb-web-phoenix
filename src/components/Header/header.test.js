import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Header from './Header';

describe('Header tests', () => {
    it ('renders Header correctly', () => {
        const tree = renderer
            .create(<Header headerLeft="left logo here" headerRight="right logo here" > Test Header </Header>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it ('triggers the onChange event when language changed', () => {
        const changleLanguageSpy = jest.spyOn(Header.prototype, 'changeLanguage'),
            wrapper = mount(<Header />);

        wrapper.find('select').simulate('change');

        expect(changleLanguageSpy).toHaveBeenCalled();
    });
});
