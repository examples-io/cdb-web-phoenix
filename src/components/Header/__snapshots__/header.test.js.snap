// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders Header correctly 1`] = `
<div
  className="app-bar"
>
  <div
    className="app-bar-left"
  >
    <img
      alt="Scotiabank"
      className={undefined}
      id="scotiabankLogo"
      src="test-file-stub"
    />
  </div>
  <div
    className="app-bar-right"
  >
    <div
      className="form--input"
    >
      <label
        className="label"
        htmlFor="languagePicker"
      >
         
         
      </label>
      <select
        className="input--select"
        id="languagePicker"
        name="language"
        onChange={[Function]}
        value="en"
      >
        <option
          value="en"
        >
           
          EN
           
        </option>
        <option
          value="fr"
        >
           
          FR
           
        </option>
      </select>
    </div>
  </div>
</div>
`;
